### ISSUE TITLE
Improve CSS link helper test to check if stylesheet loaded or not

### ISSUE THREAD

**Issue by Matthew-Bonner** [aspnet/Mvc#6619]
I have the following razor code:

    <link rel="stylesheet" href="https://cdn.evaluate.com/css/site.min.css"
                  asp-fallback-href="~/css/site.min.css"
                  asp-fallback-test-class="fallback-test" asp-fallback-test-property="table-layout" asp-fallback-test-value="auto" />

Which generates the following HTML:

    <link rel="stylesheet" href="https://cdn.evaluate.com/css/site.min.css" />
    <meta name="x-stylesheet-fallback-test" content="" class="fallback-test" />
    <script>!function(a,b,c,d){var e,f=document,g=f.getElementsByTagName("SCRIPT"),h=g[g.length-1].previousElementSibling,i=f.defaultView&&f.defaultView.getComputedStyle?f.defaultView.getComputedStyle(h):h.currentStyle;if(i&&i[a]!==b)for(e=0;e<c.length;e++)f.write('<link href="'+c[e]+'" '+d+"/>")}("table-layout","auto",["\/css\/site.min.css"], "rel=\u0022stylesheet\u0022 ");</script>

But when the CDN goes down the local fallback CSS file is not loaded.

In the network tab of the developer console in Firefox, there is no request for the local fallback.

**Comment by Matthew-Bonner** [comment]
There are no errors displayed in the console.

@NTaylorMullen @DamianEdwards

**Comment by Matthew-Bonner** [comment]
The reason for this is because if the stylesheet fails to load the browser will apply default styles, and in this case table-layout is set to auto when the file fails to load in, so this check isn't very robust.

This is the 1 liner I used:
document.defaultView.getComputedStyle(document.getElementsByTagName("SCRIPT")[1].previousSibling)["table-layout"]

and the return value was:
auto

**Comment by Eilon** [comment]
Hi @Matthew-Bonner - we're unclear about exactly what the issue is here. Would it be better to just check for the existence of a different CSS style that is different from the browser's default style? That is, should the check in your fallback be for a more "unique" style in your CDN's stylesheet?

**Comment by Matthew-Bonner** [comment]
I've already changed my code to check for the existence of a different CSS style.

My point is that plenty will be caught out by adding a class, property and value, not realising that the class will be applied to the link element.

I tried to look for an obscure property and value that I would never use in a million years, because I work with developers who would look at the class and think it is used for styling visible on a page.

Because there is no documentation saying how this feature works to the detail required to understand what is happening, we have to decipher the script injected into the page to figure out what is happening.

The simplicity of the check is great, but if another stylesheet has applied styles to the link element the check will break.

The issue is that the check to see if the CSS file loaded in or not is just well, too simple.

Because you are injecting a script into the page, can't it be a bit more clever and actually see if the file failed to load or not?
